templates/static/js/example.js

```diff
@@ -33,37 +75,81 @@
   };
 
   document.addEventListener('DOMContentLoaded', function () {
-    var examples = document.querySelectorAll('.js-example');
+    const examples = document.querySelectorAll('.js-example');
 
     [].slice.call(examples).forEach(fetchExample);
   });
 
-  function fetchExample(exampleElement) {
-    var link = exampleElement.href;
-    var request = new XMLHttpRequest();
+  /**
+   * `fetch()` wrapper that throws an error if the response is not OK.
+   * @param {String} url Address to fetch
+   * @param {RequestInit} opts Options for the fetch request
+   * @returns {Promise<Response>} Response object
+   * @throws {Error} If the response is not in the 200 (OK) range
+   */
+  const fetchResponse = async function (url, opts = {}) {
+    const response = await fetch(url, opts);
```

Comment:
The `fetchResponse` in this code branch always had a `catch` block in `fetchExample()` (I was delegating the error handling to `fetchResponse()`'s callers. I think network code like this should always throw an error on failure so that the caller can handle appropriately.

That being said, I've added an additional `try/catch` block inside here that logs an error on a network request failure, and then re-throws the error for delegated error handling by the caller.